Write a python program to implement simple Linear Regression for predicting house
price.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
import seaborn as sns

In [5]:
data=pd.read_csv('USA_Housing (1).csv')


In [8]:

x=data[['Avg. Area Income']]#(predictor)or independanat varible
y=data['Price']#target or dependat varibe
from scipy.stats import zscore

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.30,random_state=1)#split data det 70,30
from sklearn.preprocessing import StandardScaler
x_train = StandardScaler().fit_transform(x_train)
x_test = StandardScaler().fit_transform(x_test)

l=LinearRegression()
l.fit(x_train,y_train)
print("intecet",l.intercept_)
df1 = pd.DataFrame(l.coef_, x.columns, columns=['Coefficient'])
df1


intecet 1232402.8065984936


Coefficient
Avg. Area Income  222573.584309

#Holding all other features fixed, a 1 unit increase in Avg. Area Income is
associated with an increase of $222573.584309
